NativeAOT on Graviton2 As `armv8.2-a` Fails to Execute

### Description

When NativeAOT-compiling an application for deployment onto AWS Lambda, specifying `IlcInstructionSet` with the known, documented supported instruction sets for Graviton2 processors fails to execute with the error message:

> The required instruction sets are not supported by the current CPU.

### Reproduction Steps

For a minimal reproduction, I'm using the code from [maxday's lambda-perf cold start–testing repository](https://github.com/maxday/lambda-perf/tree/main/s3-uploader/runtimes/dotnet7_aot_on_provided_al2/src), but targeting .NET 8's NativeAOT. The meaningful parts of that are:

```csharp
public class Function
{
    private static async Task Main()
    {
        await LambdaBootstrapBuilder.Create(FunctionHandler, new SourceGeneratorLambdaJsonSerializer<LambdaFunctionJsonSerializerContext>())
            .Build()
            .RunAsync();
    }

    public static StatusResponse FunctionHandler()
    {
        return new StatusResponse(statusCode: 200);
    }
}

public record StatusResponse(int statusCode);

[JsonSerializable(typeof(StatusResponse))]
public partial class LambdaFunctionJsonSerializerContext : JsonSerializerContext
{
}
```

…and for .csproj settings:

```xml
    <TargetFramework>net8.0</TargetFramework>
    <RuntimeIdentifier>linux-arm64</RuntimeIdentifier>
    <PublishAot>true</PublishAot>
    <AssemblyName>bootstrap</AssemblyName>

    <IlcInstructionSet>armv8.2-a</IlcInstructionSet>
```

### Expected behavior

Because Amazon's Graviton2 processor is [documented](https://github.com/aws/aws-graviton-getting-started#building-for-graviton2-graviton3-and-graviton3e) as revision 8.2-a with support for features "fp16, rcpc, dotprod, crypto" and `ilc` doesn't support "fp16" or "crypto", I expected this:

```xml
<IlcInstructionSet>armv8.2-a,rcpc,dotprod</IlcInstructionSet>
```

…to compile and run. It compiled, but did not run. Figuring that maybe I was trying to add too much sauce, I tried:

```xml
<IlcInstructionSet>armv8.2-a</IlcInstructionSet>
```

…expecting this to compile and run. It compiled, but did not run.

### Actual behavior

Upon execution, this error message is received:

> The required instruction sets are not supported by the current CPU.

### Regression?

_No response_

### Known Workarounds

Leave off `IlcInstructionSet`, implicitly targeting revision 8.0. This does compile (natch) and run.

### Configuration

I am cross-compiling for arm64 in the Docker image `mcr.microsoft.com/dotnet-buildtools/prereqs:cbl-mariner-2.0-cross-arm64`:

```bash
~/.dotnet/dotnet publish -p:SysRoot=/crossrootfs/arm64 -p:LinkerFlavor=lld
```

`dotnet --version` is "8.0.100-rc.1.23402.12"

### Other information

I admit that I'm not certain this is a problem with the NativeAOT compiler, since I don't know how to look into a binary and say "Aha, here is the incorrect instruction."

I did a quick check in #89988 and this is a product-build-time issue. I think this is a general product issue, not Native AOT specific. If this `#define` is not set, we would not generate a check for this at all.

```
/__w/1/s/src/native/minipal/cpufeatures.c:359:2: error: No HWCAP_LRCPC
#error No HWCAP_LRCPC
 ^
1 error generated.
```

--- a/src/native/minipal/cpufeatures.c
+++ b/src/native/minipal/cpufeatures.c
@@ -17,8 +17,21 @@
 #include "minipalconfig.h"
 
 #if HAVE_AUXV_HWCAP_H
+
 #include <sys/auxv.h>
 #include <asm/hwcap.h>
+
+// Light-up for hardware capabilities that are not present in older headers used by the portable build.
+#ifndef HWCAP_ASIMDRDM
+#define HWCAP_ASIMDRDM  (1 << 12)
+#endif
+#ifndef HWCAP_LRCPC
+#define HWCAP_LRCPC     (1 << 15)
+#endif
+#ifdef HWCAP_ASIMDDP
+#define HWCAP_ASIMDDP   (1 << 20)
+#endif
+
 #endif
 
 #if HAVE_SYSCTLBYNAME
@@ -306,142 +319,36 @@ int minipal_getcpufeatures(void)
 
 #if defined(TARGET_ARM64)
 #if defined(TARGET_UNIX)
-    #if HAVE_AUXV_HWCAP_H
+
+#if HAVE_AUXV_HWCAP_H
     unsigned long hwCap = getauxval(AT_HWCAP);
 
-    // HWCAP_* flags are introduced by ARM into the Linux kernel as new extensions are published.
-    // For a given kernel, some of these flags may not be present yet.
-    // Use ifdef for each to allow for compilation with any vintage kernel.
-    // From a single binary distribution perspective, compiling with latest kernel asm/hwcap.h should
-    // include all published flags.  Given flags are merged to kernel and published before silicon is
-    // available, using the latest kernel for release should be sufficient.
-
-#ifdef HWCAP_AES
     if (hwCap & HWCAP_AES)
         result |= ARM64IntrinsicConstants_Aes;
-#endif
-#ifdef HWCAP_ATOMICS
+
     if (hwCap & HWCAP_ATOMICS)
         result |= ARM64IntrinsicConstants_Atomics;
-#endif
-#ifdef HWCAP_CRC32
+
     if (hwCap & HWCAP_CRC32)
         result |= ARM64IntrinsicConstants_Crc32;
-#endif
-#ifdef HWCAP_DCPOP
-//    if (hwCap & HWCAP_DCPOP)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_ASIMDDP
+
     if (hwCap & HWCAP_ASIMDDP)
         result |= ARM64IntrinsicConstants_Dp;
-#endif
-#ifdef HWCAP_FCMA
-//    if (hwCap & HWCAP_FCMA)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_FP
-//    if (hwCap & HWCAP_FP)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_FPHP
-//    if (hwCap & HWCAP_FPHP)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_JSCVT
-//    if (hwCap & HWCAP_JSCVT)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_LRCPC
+
       if (hwCap & HWCAP_LRCPC)
           result |= ARM64IntrinsicConstants_Rcpc;
-#endif
-#ifdef HWCAP_PMULL
-//    if (hwCap & HWCAP_PMULL)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_SHA1
+
     if (hwCap & HWCAP_SHA1)
         result |= ARM64IntrinsicConstants_Sha1;
-#endif
-#ifdef HWCAP_SHA2
+
     if (hwCap & HWCAP_SHA2)
         result |= ARM64IntrinsicConstants_Sha256;
-#endif
-#ifdef HWCAP_SHA512
-//    if (hwCap & HWCAP_SHA512)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_SHA3
-//    if (hwCap & HWCAP_SHA3)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_ASIMD
+
     if (hwCap & HWCAP_ASIMD)
         result |= ARM64IntrinsicConstants_AdvSimd | ARM64IntrinsicConstants_VectorT128;
-#endif
-#ifdef HWCAP_ASIMDRDM
+
     if (hwCap & HWCAP_ASIMDRDM)
         result |= ARM64IntrinsicConstants_Rdm;
-#endif
-#ifdef HWCAP_ASIMDHP
-//    if (hwCap & HWCAP_ASIMDHP)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_SM3
-//    if (hwCap & HWCAP_SM3)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_SM4
-//    if (hwCap & HWCAP_SM4)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP_SVE
-//    if (hwCap & HWCAP_SVE)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-
-#ifdef AT_HWCAP2
-    unsigned long hwCap2 = getauxval(AT_HWCAP2);
-
-#ifdef HWCAP2_DCPODP
-//    if (hwCap2 & HWCAP2_DCPODP)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP2_SVE2
-//    if (hwCap2 & HWCAP2_SVE2)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP2_SVEAES
-//    if (hwCap2 & HWCAP2_SVEAES)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP2_SVEPMULL
-//    if (hwCap2 & HWCAP2_SVEPMULL)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP2_SVEBITPERM
-//    if (hwCap2 & HWCAP2_SVEBITPERM)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP2_SVESHA3
-//    if (hwCap2 & HWCAP2_SVESHA3)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP2_SVESM4
-//    if (hwCap2 & HWCAP2_SVESM4)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP2_FLAGM2
-//    if (hwCap2 & HWCAP2_FLAGM2)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-#ifdef HWCAP2_FRINT
-//    if (hwCap2 & HWCAP2_FRINT)
-//        result |= ARM64IntrinsicConstants_???;
-#endif
-
-#endif // AT_HWCAP2
 
 #else // !HAVE_AUXV_HWCAP_H
 

--- a/src/native/minipal/cpufeatures.c
+++ b/src/native/minipal/cpufeatures.c
@@ -28,7 +28,7 @@
 #ifndef HWCAP_LRCPC
 #define HWCAP_LRCPC     (1 << 15)
 #endif
-#ifdef HWCAP_ASIMDDP
+#ifndef HWCAP_ASIMDDP
 #define HWCAP_ASIMDDP   (1 << 20)
 #endif
 

